### PR TITLE
Navigation component: Ensure setActiveMenu target exist

### LINKED ISSUE #25249: Navigation Component: Ensure That setActiveItem and setActiveMenu Targets Exist
When setting the `Navigation` state, we should check that the provided `item` and `menu` exist before calling the setters.

https://github.com/WordPress/gutenberg/blob/668a710b3eb9c942b21eb5913fbe304b15ea71dd/packages/components/src/navigation/index.js#L32-L41

Calling `setActiveItem` on a non-existing item is not a big deal per se (the Navigation simply won't highlight any items), but it might cause troubles with the `onActivateItem` callback exposed to the outside.

On the other hand, calling `setActiveMenu` on a non-existing menu will hide all menus.
It should make sense to default it to the `root` menu.

This would require traversing the `Navigation` children to check if the target exists, and so it depends on #25246 exploration.

### PR DESCRIPTION
**Pull request by david-szabo97**
Fixes: https://github.com/WordPress/gutenberg/issues/25249
Depends on: #25340

## Description
Check if the target of `setActiveMenu` exists.

## How has this been tested?
`yarn storybook:dev`
Added a new item in the menu called `Navigate to a non-existing menu`, when clicking this nothing should happen. Without the changes though, the whole navigation menu would disappear.

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by Copons** [comment]
Heads up: we are removing the internal `activeItem` state in #25281.
Once that's merged, the changes to `setActiveItem` in this PR won't be relevant anymore

You can read more about it here: https://github.com/WordPress/gutenberg/pull/25281#issuecomment-692812137.

tl;dr
Keeping `activeItem` internally feels great in the Storybook, but it doesn't have any real use cases.
The `activeItem` is supposed to be connected to something that happens _outside_ the component (e.g. the current page, or them template part displayed in the editor.
Since the internal and external states are always kept in sync, the internal could easily get in the way of the external.

File: packages/components/src/navigation/stories/index.js
Comment (Copons) [inline packages/components/src/navigation/stories/index.js:53]: Minor: I've noticed that long labels are displayed centered.
Would you mind adding a `text-align: left` to `ItemTitleUI` in the [navigation styles](https://github.com/WordPress/gutenberg/blob/7c467c773b7c391bfe40281a39ac5630688e3437/packages/components/src/navigation/styles/navigation-styles.js#L100-L102)?
It's unrelated to this change, but it's a one-liner that would prevent that awkwardness in the story example.

**Comment by Copons** [comment]
This looks good to me! ✨ 
Let's keep it on hold until #25340 and #25281 are merged!

File: packages/components/src/navigation/stories/index.js
Comment (david-szabo97) [reply packages/components/src/navigation/stories/index.js:53]: Yep! Noticed that one too but for some reason, I thought it's intentional or something like that :D I wanted to bring it up but totally forgot about it... Thanks!

File: packages/components/src/navigation/index.js
Comment (Copons) [inline packages/components/src/navigation/index.js:36]: Nitpick: this should be readable enough even as:
`if ( ! navigationTree.getMenu( menuId ) )`

**Comment by Copons** [comment]
@david-szabo97 Make sure to rebase and update the PR title and description (`activeItem` is not relevant anymore) before merging!

**Comment by david-szabo97** [comment]
Both #25340 and #25281 are merged! Waiting for tests and an approve 😄

**Comment by david-szabo97** [comment]
Rebased again, should be good now

**Review by Copons — APPROVED** [review]

**Pull request merged**
